### PR TITLE
Autohide the cursor in gameplay in rulesets without a cursor

### LINKED ISSUE #2901: Gameplay cursor should disappear on idle
The cursor should be hidden during osu!taiko gameplay (and osu!catch and osu!mania in the future). It only  serves as a distraction outside of clicking the skip button, and clicking the button at the bottom right of the screen. Unless the cursor is near that button, it should be hidden.

### PR DESCRIPTION
**Pull request by benpigchu**
Prereqs:
- [x] https://github.com/ppy/osu/pull/3415

- Closes #3449 
- Resolves #2901 (and maybe #3587 ?)

### CODE REVIEW

File: osu.Game/Screens/Play/Player.cs [hunk 1]
```diff
@@ -54,8 +55,10 @@ public class Player : ScreenWithBeatmapBackground, IProvideCursor
 
         public int RestartCount;
 
+        private const double cursor_idle_delay = 1000;
+
         public CursorContainer Cursor => RulesetContainer.Cursor;
-        public bool ProvidingUserCursor => RulesetContainer?.Cursor != null && !RulesetContainer.HasReplayLoaded.Value;
+        public bool ProvidingUserCursor => (RulesetContainer?.Cursor != null || idleTracker?.MouseIdleTime > cursor_idle_delay) && !RulesetContainer.HasReplayLoaded.Value;
```
Comment (peppy) [inline osu.Game/Screens/Play/Player.cs]: shouldn't this be `&&` `<=`?

Comment (benpigchu) [reply osu.Game/Screens/Play/Player.cs]: It is [have a not null cursor provided] or [(do not have a not null cursor provided) and the time that mouse is idle is longer than `cursor_idle_delay`], and the outer menu cursor should be hidden if it is not playing a replay.

**Comment by smoogipoo** [comment]
So as far as I can see this was not tested, because it doesn't work. It does not:
1) Hide the cursor when it's inactive in osu! (and the menu cursor shouldn't return if it hides).
2) Hide the cursor in a ruleset that does not have a cursor.

 The PR will also require updating with the latest master.

**Comment by peppy** [comment]
See conditional mistake as pointed out above, makes sense that it doesn't work.

**Comment by benpigchu** [comment]
@smoogipoo 
Maybe I misunderstand what is the expected behaviour of cursor...

Currently in this PR:
- The behaviour of cursor in osu!std the gameplay is not changed.
- In rulesets that is not osu!std (do not have a cursor provided), the menu cursor will disappear if the user do not have mouse input for a while (1 second). (maybe you will say that the cursor is not hiding if you do not wait for 1 second or move the cursor)

What is wanted seems to be:
- In osu!std the cursor should hide when the cursor is "inactive" (no mouse input for a while, or plus in a break time?)
- In rulesets that is not osu!std (do not have a cursor provided), the menu cursor should just disappear (or only appear when in break time and has mouse input recently?)

I will change my implement if this is clarified.

**Comment by benpigchu** [comment]
#3415 is reverted recently by #3737 ...
I think I should wait for #3739

**Comment by benpigchu** [comment]
Let me close this for now

**Pull request closed without merging**